src/hooks/utils/useDialArchaeologists.ts

```diff
@@ -33,7 +36,33 @@ export function useDialArchaeologists(setIsDialing: React.Dispatch<React.SetStat
     [libp2pNode, setIsDialing, toast]
   );
 
+  const pingArchaeologist = useCallback(async (peerId: PeerId, onComplete: Function) => {
+    const pingTimeout = 5000;
+
+    const couldNotConnect = setTimeout(() => {
+      console.log('ping timeout!');
+
+      dispatch(setArchaeologistException(
+        peerId.toString(),
+        { code: ArchaeologistExceptionCode.CONNECTION_EXCEPTION, message: 'Ping timeout' }
+      ));
+      onComplete();
+    }, pingTimeout);
+
+    console.log('pinging...');
+
+    const latency = await libp2pNode?.ping(peerId);
+    console.log('latency: ', latency);
+
+
+    if (!!latency) {
+      clearTimeout(couldNotConnect);
+      onComplete();
+    }
+  }, [libp2pNode, dispatch]);
```

Comment:
This file got moved in the create-sarcophagus-refactor PR to a different folder, so there will be a merge conflict -- this new method just needs added to file in the new location.